odl/operator/pspace_ops.py

```diff
@@ -203,50 +203,21 @@ def __init__(self, operators, dom=None, ran=None):
 
         super().__init__(domain=dom, range=ran, linear=linear)
 
-    def _apply(self, x, out):
-        """Call the ProductSpace operators in-place.
+    def _call(self, x, out=None):
+        """Call the ProductSpace operators.
 
         Parameters
         ----------
         x : domain element
             input vector to be evaluated
-        out : range element
+        out : range element, optional
             output vector to write result to
 
         Returns
         -------
-        `None`
-
-        Examples
-        --------
-        See `_call`
-        """
-        has_evaluated_row = np.zeros(self.range.size, dtype=bool)
-        for i, j, op in zip(self.ops.row, self.ops.col, self.ops.data):
-            if not has_evaluated_row[i]:
-                op(x[j], out=out[i])
-            else:
-                # TODO: optimize
-                out[i] += op(x[j])
-
-            has_evaluated_row[i] = True
-
-        for i, evaluated in enumerate(has_evaluated_row):
-            if not evaluated:
-                out[i].set_zero()
-
-    def _call(self, x):
-        """Call the ProductSpace operators.
-
-        Parameters
-        ----------
-        x : domain element
-            Input vector to be evaluated
-
-        Returns
-        -------
         out : range element
-            Result of the evaluation
+            Result of the evaluation. If ``out`` was provided, the
+            returned object is identical with it.
```

Comment:
"Identity" is not the same as "Identical", for me, identical has a very strong notation for things "looking the same", say "identical twins", for example. It seems numpy, where they document `out`, write something like 

> If out is provided, the function writes the result into it, and returns a reference to out. (See Examples)

[np.cos](http://docs.scipy.org/doc/numpy-1.10.0/reference/generated/numpy.cos.html)

Perhaps we could write "If out was provided, the returned object is a reference to it"..